bulk-model-sync-mps/src/main/kotlin/org/modelix/mps/model/sync/bulk/MPSBulkSynchronizer.kt

```diff
@@ -141,6 +151,118 @@ object MPSBulkSynchronizer {
         }
     }
 
+    @OptIn(ExperimentalSerializationApi::class)
+    @JvmStatic
+    fun importRepositoryFromModelServer() {
+        val repository = getRepository()
+
+        val includedModuleNames = parseRawPropertySet(System.getProperty("modelix.mps.model.sync.bulk.input.modules"))
+        val includedModulePrefixes = parseRawPropertySet(System.getProperty("modelix.mps.model.sync.bulk.input.modules.prefixes"))
+        val continueOnError = System.getProperty("modelix.mps.model.sync.bulk.input.continueOnError", "false").toBoolean()
+
+        val modelServerUrl = System.getProperty("modelix.mps.model.sync.bulk.server.url")
+        val repositoryId = RepositoryId(requireNotNull(System.getProperty("modelix.mps.model.sync.bulk.server.repository")) { "modelix.mps.model.sync.bulk.server.repository not specified" })
+        val branchName = System.getProperty("modelix.mps.model.sync.bulk.server.branch")
+        val branchRef = repositoryId.getBranchReference(branchName)
+        val versionHash: String? = System.getProperty("modelix.mps.model.sync.bulk.server.version.hash")
+        val baseVersionHash: String? = System.getProperty("modelix.mps.model.sync.bulk.server.version.base.hash")
+        val client = ModelClientV2.builder().url(modelServerUrl).build()
+        val version = runBlocking {
+            if (versionHash == null) client.lazyLoadVersion(branchRef) else client.lazyLoadVersion(repositoryId, versionHash)
+        }
+        val baseVersion = baseVersionHash?.let {
+            runBlocking {
+                client.lazyLoadVersion(repositoryId, it)
+            }
+        }
+        println("Loading version ${version.getContentHash()}")
+
+        val access = repository.modelAccess
+        access.executeCommandInEDT {
+            if (baseVersion != null) {
+                val invalidationTree = InvalidationTree(1_000_000)
+                val newTree = version.getTree()
+                fun ancestorsAsArray(nodeId: Long) = newTree.ancestorsAndSelf(nodeId).toList().asReversed().toLongArray()
+                newTree.visitChanges(baseVersion.getTree(), object : ITreeChangeVisitorEx {
```

Comment:
Can we put this into a standalone class to reduce the size of this method? That also opens the possibility to properly name that class with a name that explains the purpose.